Dockerfile

```diff
@@ -0,0 +1,15 @@
+# syntax=docker/dockerfile:1
+
+FROM golang:1.21
+
+COPY . /root
+WORKDIR /root
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux go build -o /durabletask-go
+
+EXPOSE 4001
+
+# Run
+ENTRYPOINT [ "/durabletask-go" ]
+CMD [ "--ip", "0.0.0.0" ]
```

Comment:
Should we make the port number explicit here too since we're hardcoding `EXPOSE 4001`?